pkg/s3presign/sts.go

```
@@ -0,0 +1,121 @@
+	}
+
+	endpoint := c.Endpoint
+	if endpoint == "" {
+		endpoint = defaultSTSEndpoint
+	}
+
+	resp, err := c.HTTPClient.PostForm(endpoint, params)
+	if err != nil {
+		return nil, fmt.Errorf("STS request failed: %w", err)
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.11.4)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)